sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClient.java

```diff
@@ -282,10 +285,18 @@ private static BiFunction<HttpClientResponse, Connection, Publisher<HttpResponse
     }
 
     /*
-     * Adds write timeout handler once the request is ready to begin sending.
+     * Adds request handlers:
+     * - write timeout handler once the request is ready to begin sending.
+     * - progress handler if progress tracking has been requested.
      */
-    private static void addWriteTimeoutHandler(Connection connection, long timeoutMillis) {
-        connection.addHandlerLast(WriteTimeoutHandler.HANDLER_NAME, new WriteTimeoutHandler(timeoutMillis));
+    private void addRequestHandlers(Connection connection, Context context) {
+        connection.addHandlerLast(WriteTimeoutHandler.HANDLER_NAME, new WriteTimeoutHandler(writeTimeout));
+        ProgressReporter progressReporter = Contexts.with(context).getProgressReporter();
```

Comment:
Should we clarify what determines if something that is user-settable via a Context becomes elevated to be an API on Contexts? For example, I see above code that gets the AZURE_RESPONSE_TIMEOUT and AZURE_EAGERLY_READ_RESPONSE values out of Context. At what stage do we do this elevation, and then, does progress reporter meet that criteria?

Reply:
I'd say that stuff that has to cross core-sdks boundary is a candidate where HLC logic has interest in orchestrating it.
Especially when there's N:1 relationship of interest, i.e. N SDKS need to use it with core (core is 1 here).
For example:
- progress reporting is present in 3 storage services plus comms. We should not use magic key for this.
- The AZURE_EAGERLY_READ_RESPONSE is a contract between RestProxy and HttpClients. Doesn't cross core boundary. Shouldn't be API
- AZURE_RESPONSE_TIMEOUT (or some form of timeout tbd). We have sync APIs that take `Duration timeout` in storage, tables. This needs to (plus-minus math transformations in retry policy) travel across sdks-core boundary. This is a good candidate for API.